First edit of release notes  for 6.10

GuiMacielPereira opened the pull request:
### Description of work
Went over release notes and converted passive voice into active voice, added images and added links to documentation pages.

### To test:

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->
Build documentation and read over release notes for 6.10.
Check that meaning of bullet points in the release notes makes sense. Check for grammatical and formatting errors.

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

GuiMacielPereira committed:
Run release_editor.py script

File changes from running release_editor.py script

GuiMacielPereira committed:
Edit the Framework page of release notes

- Added gif showing capability of instrument data cache
- Added image for Filter Events GUI
- Changed notes to active voice
- Corrected formatting errors

GuiMacielPereira committed:
Edit mantidworkbench release notes

- Changed to active voice
- Added images

GuiMacielPereira committed:
Edit Diffraction page

- Added label to gsas section of engineering diffraction documentation
- Switch passive voice to active voice in release notes

GuiMacielPereira committed:
Edit Muon page

- Added links and made change from passive to active voice

GuiMacielPereira committed:
Edit Reflectometry page

Added image for loading ORSO Nexus files

GuiMacielPereira committed:
Edit SANS page

- Added picture of new algorithm added

GuiMacielPereira committed:
Edit Direct Geometry page

- Added hyperlink

GuiMacielPereira committed:
Edit indirect diffraction interface

- Passive to active voice
- Deleted repeated lines
- Added image

GuiMacielPereira committed:
Edit the Inelastic page

- Added figures
- Changed passive to active voice

jhaigh0 commented on docs/source/release/v6.10.0/diffraction.rst:
```diff
@@ -10,34 +10,53 @@ Powder Diffraction
 Bugfixes
 ############
-.. amalgamate:: Diffraction/Powder/Bugfixes
+- HRPD reduction :ref:`isis-powder-diffraction-hrpd-ref` now only subtracts empty run from the vanadium when it is subtracted from the sample run (i.e. when ``subtract_empty_instrument=True``) - note other ISIS instrument will always subtract an empty run from the vanadium.
```
Might make more sense when rendered.
Currently:
![image](https://github.com/mantidproject/mantid/assets/35813666/94b4f226-2d51-4c80-b567-f26dc5d701d2)

```suggestion
- :ref:`HRPD reduction script <isis-powder-diffraction-hrpd-ref>` now only subtracts empty run from the vanadium when it is subtracted from the sample run (i.e. when ``subtract_empty_instrument=True``) - note other ISIS instrument will always subtract an empty run from the vanadium.
```

jhaigh0 commented on docs/source/release/v6.10.0/direct_geometry.rst:
```diff
@@ -10,35 +10,11 @@ General
 Bugfixes
 ############
-.. amalgamate:: Direct_Geometry/General/Bugfixes
+- Fixed a bug that prevented to show detectors for Let data.
```
```suggestion
- Fixed a bug that prevented being able to `show detectors` for Let data.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
 
 New features
 ############
-.. amalgamate:: Framework/Algorithms/New_features
+- Algorithm :ref:`algm-SaveASCII` now supports 1D :ref:`MDHistoWorkspace`.
+- Algorithm :ref:`algm-CompressEvents` is now able to combine events together in logarithmic increasing size groups.
+- Algorithm `PolarizationCorrectionWildes` now accepts input workspace order.
+- Algorithms :ref:`algm-Abins` and :ref:`algm-Abins2D` can now import phonopy .yaml data.
+  The force constants are stored in a file named ``FORCE_CONSTANTS`` or
+  ``force_constants.hdf5`` in the same directory as the YAML file.
+  This is recommended when using large force constants arrays as the
+  YAML loader can be very slow.
```
The spacing of the bullet points in this section is wider once built. Not a critical issue, could be to do with the 
```
The Euphonic JSON formats are convenient to create with Python
scripts, and recommended for users who wish to somehow customise or
manipulate their data before using it with Abins(2D).
```
text block?

GuiMacielPereira replied:
Yeah, it was caused by that extra indentation

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
 ############
-.. amalgamate:: Framework/Algorithms/New_features
+- Algorithm :ref:`algm-SaveASCII` now supports 1D :ref:`MDHistoWorkspace`.
+- Algorithm :ref:`algm-CompressEvents` is now able to combine events together in logarithmic increasing size groups.
```
```suggestion
- Algorithm :ref:`algm-CompressEvents` is now able to combine events together in logarithmically increasing size groups.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
-.. amalgamate:: Framework/Algorithms/New_features
+- Algorithm :ref:`algm-SaveASCII` now supports 1D :ref:`MDHistoWorkspace`.
+- Algorithm :ref:`algm-CompressEvents` is now able to combine events together in logarithmic increasing size groups.
+- Algorithm `PolarizationCorrectionWildes` now accepts input workspace order.
```
```suggestion
- Algorithm :ref:`algm-PolarizationCorrectionWildes` now accepts input workspace order.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
+  scripts, and recommended for users who wish to somehow customise or
+  manipulate their data before using it with Abins(2D).
+
+- New algorithm :ref:`LoadErrorEventsNexus <algm-LoadErrorEventsNexus>` to load events from the `bank_error_events` bank of a NeXus file.
```
```suggestion
- New algorithm :ref:`LoadErrorEventsNexus <algm-LoadErrorEventsNexus>` to load events from the ``bank_error_events`` bank of a NeXus file.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
+- Algorithm :ref:`FindSXPeaks <algm-FindSXPeaks-v1>` supports new peak finding strategy ``AllPeaksNSigma``.
+  Credits to the author of SXD2001 for the idea of using NSigma as a threshold (albeit in SXD2001 the peak finding is done in 3D).
+  Gutmann, M. J. (2005). SXD2001. ISIS Facility, Rutherford Appleton Laboratory, Oxfordshire, England.
+- Algorithm :ref:`FindSXPeaks <algm-FindSXPeaks-v1>` now includes validation rules to remove spurious peaks due to noise,
+  by allowing user to provide additional arguements as below:
+  - ``MinNBinsPerPeak``, the Minimum number of bins contributing to a peak in an individual spectrum
+  - ``MinNSpectraPerPeak``, ``MaxNSpectraPerPeak`` Minimum & Maximum number of spectra contributing to a peak after they are grouped.
```
```suggestion
- Algorithm :ref:`FindSXPeaks <algm-FindSXPeaks-v1>` now includes validation rules to remove spurious peaks due to noise,
  by allowing user to provide additional arguements as below:

  - ``MinNBinsPerPeak``, the Minimum number of bins contributing to a peak in an individual spectrum
  - ``MinNSpectraPerPeak``, ``MaxNSpectraPerPeak`` Minimum & Maximum number of spectra contributing to a peak after they are grouped.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
 ############
-.. amalgamate:: Framework/Algorithms/Bugfixes
+- Algorithm :ref:`LoadNexusProcessed <algm-LoadNexusProcessed>` is now faster to load a NeXus file.
+- Algorithm ``DSFinterp``, which was deprecated, has been removed
```
```suggestion
- Algorithm ``DSFinterp``, which was deprecated, has been removed.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
+    behaviour.
+
+- Algortihm :ref:`Load <algm-Load>` now loads a single file faster.
+- Fixed doctest strings for :ref:`GenerateGroupingPowder <algm-GenerateGroupingPowder>`.
```
Not sure this needs to be in the release notes?

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
 ############
-.. amalgamate:: Framework/Fit_Functions/Bugfixes
+- Search box for fitting functions in Fit interface no longer shows duplicate functions.
+- Fit Function DSFinterp1DFit, which was deprecated, has been removed.
```
```suggestion
- Fit Function ``DSFinterp1DFit``, which was deprecated, has been removed.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
-.. amalgamate:: Framework/Fit_Functions/Bugfixes
+- Search box for fitting functions in Fit interface no longer shows duplicate functions.
+- Fit Function DSFinterp1DFit, which was deprecated, has been removed.
+- Function :ref:`func-CompositeFunction` will now throw an exception if ``getNumberDomains()`` is called and there is an inconsistent number of domains in any of the member functions.
```
```suggestion
- :ref:`func-CompositeFunction` will now throw an exception if ``getNumberDomains()`` is called and there is an inconsistent number of domains in any of the member functions.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
-.. amalgamate:: Framework/Data_Handling/New_features
+- Algorithm :ref:`algm-LoadEventAsWorkspace2D` accepts new boolean parameter ``LoadNexusInstrumentXML``. Default is *true*.
+- File search/loading will now look in instrument data cache **on IDAaaS**.
+  The instrument data cache is the directory ``/data/instrument/`` present on IDAaaS, and contains a local copy of the data archive.
```
I don't think it's quite correct that it "contains a local copy of the data archive", is it just the most recent three years worth of data?

GuiMacielPereira replied:
Yeah, you're right, I will specify that

jhaigh0 replied:
I think you've got the same bullet point spacing error here as you had in the workbench docs, so probably something to do with the figure again.

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
 ############
-.. amalgamate:: Framework/Data_Handling/Bugfixes
+- Algorithm :ref:`algm-Load-v1` now guarantees that properties ``LoaderName`` and ``LoaderVersion`` are set by end of algorithm.
+- Algorithm :ref:`algm-GenerateGroupingPowder` now allows that just one of the properties ``GroupingWorkspace`` and ``GroupingFilename`` be set.
```
```suggestion
- Algorithm :ref:`algm-GenerateGroupingPowder` now allows just one of the properties ``GroupingWorkspace`` and ``GroupingFilename`` be set.
```

jhaigh0 commented on docs/source/release/v6.10.0/framework.rst:
```diff
@@ -10,70 +10,128 @@ Algorithms
-Bugfixes
-############
-.. amalgamate:: Framework/Data_Objects/Bugfixes
+- Speedup processing of IDF XML during loading when side-by-side-view-location parameter is not used.
```
```suggestion
- Sped up processing of IDF XML during loading when ``side-by-side-view-location`` parameter is not used.
```

jhaigh0 commented on docs/source/release/v6.10.0/indirect_geometry.rst:
```diff
@@ -7,23 +7,32 @@ Indirect Geometry Changes
 New Features
 ------------
-.. amalgamate:: Indirect/New_features
+- Algorithm :ref:`algm-ISISIndirectEnergyTransfer` now has sample logs for multiple run reductions. ``run_number`` log stores the run numbers used in the reduction, separated by commas. ``multi_run_reduction`` log informs about the reduction done on chopped runs or regular runs.
```
```suggestion
- Algorithm :ref:`algm-ISISIndirectEnergyTransfer` now has sample logs for multiple run reductions. The ``run_number`` log stores the run numbers used in the reduction, separated by commas. The ``multi_run_reduction`` log informs about the reduction done on chopped runs or regular runs.
```

jhaigh0 commented on docs/source/release/v6.10.0/inelastic.rst:
```diff
@@ -7,23 +7,56 @@ Inelastic Changes
 New features
 ############
-.. amalgamate:: Inelastic/Algorithms/New_features
+- Algorithm `TransformToIqt` and `CalculateIqt` has checkbox with title ``EnfoceNormalization``. This option is set to ``True`` by default (no change to the current algorithms/workflow occurs). When it is set to ``False``, the ``LHSWorkspace`` from the output from ``ExtractFFTSpectrum`` is used in both branches to perform the final workspace division and the two intermediate workspace divisions a ...
```
```suggestion
- Algorithm :ref:`algm-TransformToIqt` and :ref:`algm-CalculateIqt` has checkbox with title ``EnfoceNormalization``. This option is set to ``True`` by default (no change to the current algorithms/workflow occurs). When it is set to ``False``, the ``LHSWorkspace`` from the output from ``ExtractFFTSpectrum`` is used in both branches to perform the final workspace division and the two intermediate workspace divisions are skipped.
```

jhaigh0 commented on docs/source/release/v6.10.0/mantidworkbench.rst:
```diff
@@ -5,38 +5,54 @@ Mantid Workbench Changes
 --------
-.. amalgamate:: Workbench/Bugfixes
+
+- Selecting or deselecting plot curves in plot settings no longer causes an error.
+- Overplotting workspaces onto a regular Matplotlib plot no longer causes an error.
+- Dragging a table workspace onto a plot no longer causes a crash.
+- Loading files from disk no longer causes a memory leak.
+- Previously hidden plots are no longer unhidden from some plot scripts.
```
The bullet point spacing here has also gone larger than normal

jhaigh0 commented on docs/source/release/v6.10.0/mantidworkbench.rst:
```diff
@@ -5,38 +5,54 @@ Mantid Workbench Changes
+- Loading files from disk no longer causes a memory leak.
+- Previously hidden plots are no longer unhidden from some plot scripts.
+- Plot scripts in documentation now use ``fig.show()`` instead of ``plt.show()`` to avoid hanging Workbench.
+- Fitting property browser no longer crashesk when removing a function from a nested composite function.
```
```suggestion
- Fit property browser no longer crashes when removing a function from a nested composite function.
```

jhaigh0 commented on docs/source/release/v6.10.0/mantidworkbench.rst:
```diff
@@ -5,38 +5,54 @@ Mantid Workbench Changes
+- Previously hidden plots are no longer unhidden from some plot scripts.
+- Plot scripts in documentation now use ``fig.show()`` instead of ``plt.show()`` to avoid hanging Workbench.
+- Fitting property browser no longer crashesk when removing a function from a nested composite function.
+- Renaming a plot in the plots widget no longer redraws the plot title regardless of the Show Title setting.
```
```suggestion
- Renaming a plot in the plots widget no longer redraws the plot title regardless of the ``Show Title`` workbench setting.
```

jhaigh0 commented on docs/source/release/v6.10.0/reflectometry.rst:
```diff
@@ -7,10 +7,17 @@ Reflectometry Changes
+   :width: 300px
+
+- Algorithm :ref:`algm-SaveISISReflectometryORSO` now outputs the individual angle and transmission files, flood correction files, calibration files and the reduction script in the ORSO file metadata.
+- Algorithm :ref:`algm-SaveISISReflectometryORSO` is now able to save multiple datasets to a single file. Note that as part of this change the InputWorkspace parameter has been renamed to WorkspaceList and takes a comma separated list of workspace names. Existing scripts that call the algorithm will need to be updated to use this new parameter name. The :ref:`Save ASCII tab<refl_save_ascii>` has  ...
```
```suggestion
- Algorithm :ref:`algm-SaveISISReflectometryORSO` is now able to save multiple datasets to a single file. Note that as part of this change the InputWorkspace parameter has been renamed to WorkspaceList and takes a comma separated list of workspace names. Existing scripts that call the algorithm will need to be updated to use this new parameter name. The :ref:`Save tab<refl_save>` has been updated with additional checkbox options to support this new functionality when saving in the ORSO format.
```

jhaigh0 reviewed (commented):
Just minor corrections. Don't commit them all individually if you want to use the git suggestions thing. If you go to the files tab you can add them all to one batch (commit).

All in all looking very nice.

GuiMacielPereira committed:
Apply suggestions from code review

- Added missing links
- Typos and grammatical corrections

Co-authored-by: Jonathan Haigh <jonathan.haigh@stfc.ac.uk>

Co-authored-by: Jonathan Haigh <35813666+jhaigh0@users.noreply.github.com>

GuiMacielPereira committed:
Fix overspacing between bullet points

In both cases, this was caused by an unexpected identation.

GuiMacielPereira committed:
Reword data cache explanation + fix bullet spacing

Reworded description of data cache to include information that only data from the past 3 years is included.
Fix spacing of bullet points by de-identing the figures

GuiMacielPereira committed:
Remove point about doctest strings

Deleted because not relevant to users

jhaigh0 reviewed (approved):
Looking good now.
To come will be a PR writing the index file overview of this version, and one to pick up any release notes introduced during the release period.
